### ISSUE TITLE
Async test method must have non-generic Task return type when no result is expected

### ISSUE THREAD

**Issue by mikkelbu** [nunit/nunit.analyzers#48]
We should warn about the following problem

**Message: Async test method must have non-generic Task return type when no result is expected**
```csharp
    [Test]
    public async Task<int> AsyncGenericTask()
    {
      return await Task.FromResult(1);
    }
```
and
```csharp
    [Test]
    public Task<int> NonAsyncGenericTask()
    {
      return Task.FromResult(1);
    }
```
and
**
